platform/ios/display_server_ios.mm

```diff
@@ -109,18 +109,39 @@
 #endif
 	if (rendering_context) {
 		if (rendering_context->initialize() != OK) {
-			memdelete(rendering_context);
-			rendering_context = nullptr;
+			bool failed = true;
```

Comment:
Removing the `memdelete`, if Vulkan and `fallback_to_vulkan` is enabled, will result in a memory leak.

Simplest approach is to leave both lines here to ensure proper cleanup.